version: '2.4'
services:
        postgres:
                build: ./postgres-docker
                container_name: mythic_postgres
                network_mode: host
                volumes:
                 - ./postgres-docker/database:/var/lib/postgresql/data
                labels:
                        NAME: "mythic_postgres"
                restart: on-failure
                environment:
                        POSTGRES_PASSWORD: "${POSTGRES_PASSWORD}"
        rabbitmq:
                build: ./rabbitmq-docker
                container_name: mythic_rabbitmq
                network_mode: host
                volumes:
                 - ./rabbitmq-docker/storage:/var/lib/rabbitmq
                labels:
                        NAME: "mythic_rabbitmq"
                restart: on-failure
                #mem_limit: 512M
        mythic:
                build: ./mythic-docker
                network_mode: host
                container_name: mythic_server
                volumes:
                 - ./mythic-docker:/Mythic
                labels:
                        NAME: "mythic_server"
                environment:
                       POSTGRES_PASSWORD: "${POSTGRES_PASSWORD}"
                depends_on:
                 - postgres
                 - rabbitmq
                restart: on-failure
                command: ["./wait-for-postgres.sh", "127.0.0.1:5432", "--", "python", "/Mythic/server.py"]

